File: ritesh/LEETCODE/Question5.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        original = x
        reverse = 0

        while x > 0:
            digit = x % 10
            reverse = reverse * 10 + digit
            x //= 10

        return original == reverse

obj = Solution()
print(obj.isPalindrome(121))    # True
print(obj.isPalindrome(-121))   # False
print(obj.isPalindrome(10))     # False

File: nyasha/leetcode/question5.java
import java.util.Scanner;

public class question5 {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = scanner.nextInt();

        int original = num;
        int reverse = 0;

        if (num < 0) {
            System.out.println("Not a Palindrome");
            scanner.close();
            return;
        }

        while (num != 0) {
            int digit = num % 10;
            reverse = reverse * 10 + digit;
            num = num / 10;
        }

        if (original == reverse) {
            System.out.println("Palindrome");
        } else {
            System.out.println("Not a Palindrome");
        }

        scanner.close();
    }
}